Source/JavaScriptCore/runtime/ObjectConstructor.cpp

```diff
@@ -656,44 +656,106 @@ bool toPropertyDescriptor(JSGlobalObject* globalObject, JSValue in, PropertyDesc
     }
     JSObject* description = asObject(in);
 
-    JSValue enumerable = description->getIfPropertyExists(globalObject, vm.propertyNames->enumerable);
-    RETURN_IF_EXCEPTION(scope, false);
-    if (enumerable)
-        desc.setEnumerable(enumerable.toBoolean(globalObject));
-
-    JSValue configurable = description->getIfPropertyExists(globalObject, vm.propertyNames->configurable);
-    RETURN_IF_EXCEPTION(scope, false);
-    if (configurable)
-        desc.setConfigurable(configurable.toBoolean(globalObject));
+    JSValue enumerable;
+    JSValue configurable;
+    JSValue value;
+    JSValue writable;
+    JSValue get;
+    JSValue set;
 
-    JSValue value = description->getIfPropertyExists(globalObject, vm.propertyNames->value);
-    RETURN_IF_EXCEPTION(scope, false);
-    if (value)
-        desc.setValue(value);
+    if (globalObject->propertyDescriptorFastPathWatchpointSet().state() == ClearWatchpoint)
+        globalObject->tryInstallPropertyDescriptorFastPathWatchpoint();
 
-    JSValue writable = description->getIfPropertyExists(globalObject, vm.propertyNames->writable);
-    RETURN_IF_EXCEPTION(scope, false);
-    if (writable)
-        desc.setWritable(writable.toBoolean(globalObject));
 
-    JSValue get = description->getIfPropertyExists(globalObject, vm.propertyNames->get);
-    RETURN_IF_EXCEPTION(scope, false);
-    if (get) {
-        if (!get.isUndefined() && !get.isCallable()) {
-            throwTypeError(globalObject, scope, "Getter must be a function."_s);
-            return false;
+    bool canUseFastPath = false;
+    if (globalObject->propertyDescriptorFastPathWatchpointSet().isStillValid() && globalObject->objectPrototypeChainIsSane() && description->inherits<JSFinalObject>() && description->getPrototypeDirect() == globalObject->objectPrototype() && description->structure()->canPerformFastPropertyEnumeration() && !hasIndexedProperties(description->indexingType())) {
+        if (!description->staticPropertiesReified()) {
```

Comment:
nit: It's extremely unlikely for a final object to have non-reified static properties. If we remove this condition, `fastForEachPropertyWithSideEffectFreeFunctor` would correctly handle such an unlikely case by returning `false`.